hazelcast/src/test/java/com/hazelcast/config/XmlOnlyConfigBuilderTest.java

```diff
@@ -0,0 +1,198 @@
+ * YAML configuration to allow usage of XML-native constructs.
+ *
+ * @see XMLConfigBuilderTest
+ */
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelTest.class})
+public class XmlOnlyConfigBuilderTest {
+
+    private static final String HAZELCAST_START_TAG = "<hazelcast xmlns=\"http://www.hazelcast.com/schema/config\">\n";
+    private static final String HAZELCAST_END_TAG = "</hazelcast>\n";
+
+    @Test(expected = InvalidConfigurationException.class)
+    public void testMissingNamespace() {
+        String xml = "<hazelcast/>";
+        buildConfig(xml);
+    }
+
+    @Test(expected = InvalidConfigurationException.class)
+    public void testInvalidNamespace() {
+        String xml = "<hazelcast xmlns=\"http://foo.bar\"/>";
+        buildConfig(xml);
+    }
+
+    @Test
+    public void testValidNamespace() {
+        String xml = HAZELCAST_START_TAG + HAZELCAST_END_TAG;
+        buildConfig(xml);
+    }
+
+    @Test(expected = InvalidConfigurationException.class)
+    public void testHazelcastTagAppearsTwice() {
+        String xml = HAZELCAST_START_TAG + "<hazelcast/>" + HAZELCAST_END_TAG;
+        buildConfig(xml);
+    }
+
+    @Test(expected = InvalidConfigurationException.class)
+    public void testHazelcastInstanceNameEmpty() {
+        String xml = HAZELCAST_START_TAG + "<instance-name></instance-name>" + HAZELCAST_END_TAG;
+        buildConfig(xml);
+    }
+
+    @Test
+    public void testXsdVersion() {
+        String origVersionOverride = System.getProperty(HAZELCAST_INTERNAL_OVERRIDE_VERSION);
+        assertXsdVersion("0.0", "0.0");
+        assertXsdVersion("3.9", "3.9");
+        assertXsdVersion("3.9-SNAPSHOT", "3.9");
+        assertXsdVersion("3.9.1-SNAPSHOT", "3.9");
+        assertXsdVersion("3.10", "3.10");
+        assertXsdVersion("3.10-SNAPSHOT", "3.10");
+        assertXsdVersion("3.10.1-SNAPSHOT", "3.10");
+        assertXsdVersion("99.99.99", "99.99");
+        assertXsdVersion("99.99.99-SNAPSHOT", "99.99");
+        assertXsdVersion("99.99.99-Beta", "99.99");
+        assertXsdVersion("99.99.99-Beta-SNAPSHOT", "99.99");
+        if (origVersionOverride != null) {
+            System.setProperty(HAZELCAST_INTERNAL_OVERRIDE_VERSION, origVersionOverride);
+        } else {
+            System.clearProperty(HAZELCAST_INTERNAL_OVERRIDE_VERSION);
+        }
+    }
+
+    @Test
+    public void testConfig2Xml2DefaultConfig() {
+        testConfig2Xml2Config("hazelcast-default.xml");
+    }
+
+    @Test
+    public void testConfig2Xml2FullConfig() {
+        testConfig2Xml2Config("hazelcast-fullconfig.xml");
+    }
+
+    @Test
+    public void testXSDDefaultXML() throws Exception {
+        testXSDConfigXML("hazelcast-default.xml");
+    }
+
+    @Test
+    public void testFullConfig() throws Exception {
+        testXSDConfigXML("hazelcast-fullconfig.xml");
+    }
+
+    // No YAML identical to the XML can be written
```

Comment:
Super minor: I think this comment is not helpful. All these tests are related to the cases that are not valid for YAML. And if at some point you add another language, for example, JSON, you'd need to maintain this comment and change to: "// No YAML or JSON identical to the XML can be written".

Reply:
Removed. I moved this working comment together with the method :) Same for the next one.